src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -79,9 +83,51 @@ public async Task RunBatchAsync(QualificationRunOptions runOptions)
          _logger.AddInfo($"Qualification scenario finished in {timeSpent.ToDisplay()}");
       }
 
-      private Task<SaticFiles> copyStaticFiles(dynamic qualificationPlan)
+      private Task updateProjectsFullPath(IReadOnlyList<Project> projects) => Task.WhenAll(projects.Select(updateProjectFullPath));
+
+      private async Task<string> downloadRemoteFile(string url, string fileName, string subFolder, string id, string type)
       {
-         var staticFiles = new SaticFiles();
+         _logger.AddDebug($"Downloading {type.ToLower()} file for {id}...");
+         var downloadFolder = Path.Combine(_runOptions.TempFolder, subFolder);
+         DirectoryHelper.CreateDirectory(downloadFolder);
+
+         var fileFullPath = Path.Combine(downloadFolder, fileName);
+
+         using (var wc = new WebClient())
+         {
+            try
+            {
+               await wc.DownloadFileTaskAsync(url, fileFullPath);
+               _logger.AddDebug($"{type} file for {id} downloaded to ${fileFullPath}");
+               return fileFullPath;
+            }
+            catch (Exception e)
+            {
+               //Exception is thrown for example if the given url does not exist or if internet access is not possible etc..
+               _logger.AddError(e.Message);
+               return url;
+            }
+         }
+      }
+
+      private async Task updateProjectFullPath(Project project)
+      {
+         Task<string> downloadRemoteProject() => downloadRemoteFile(project.Path, $"{project.Id}{Filter.JSON_EXTENSION}", PROJECT_DOWNLOAD_FOLDER, project.Id, "Project");

```

Comment:
`downloadRemoteFile(..., $"{project.Id}{Filter.JSON_EXTENSION}, ..."`

I wouldn't use _projectId_ as download file name. Ids could theoretically contain characters invalid for path.
Instead I would use the original file name.
E.g. if the url is `https://blabla.com/Tralala.json` then downloaded file name should be `Tralala.json`